File: ICBORNumber.cs
/*
Written in 2013 by Peter O.
Any copyright is dedicated to the Public Domain.
http://creativecommons.org/publicdomain/zero/1.0/
If you like this, you should donate to Peter O.
at: http://peteroupc.github.io/CBOR/
 */
using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
// using System.Numerics;
using System.Text;

namespace PeterO
{
	internal interface ICBORNumber
	{
		bool IsPositiveInfinity(Object obj);
		bool IsInfinity(Object obj);
		bool IsNegativeInfinity(Object obj);
		bool IsNaN(Object obj);
		double AsDouble(Object obj);
		ExtendedDecimal AsExtendedDecimal(Object obj);
		ExtendedFloat AsExtendedFloat(Object obj);
		float AsSingle(Object obj);
		BigInteger AsBigInteger(Object obj);
		long AsInt64(Object obj);
		bool CanFitInSingle(Object obj);
		bool CanFitInDouble(Object obj);
		bool CanFitInInt32(Object obj);
		bool CanFitInInt64(Object obj);
		bool CanFitInTypeZeroOrOne(Object obj);
		bool CanTruncatedIntFitInInt64(Object obj);
		bool CanTruncatedIntFitInInt32(Object obj);
		int AsInt32(Object obj, int minValue, int maxValue);
		bool IsZero(Object obj);
		int Sign(Object obj);
		bool IsIntegral(Object obj);
	}
}

File: src/main/java/com/upokecenter/cbor/ICBORNumber.java
package com.upokecenter.cbor;
/*
Written in 2013 by Peter O.
Any copyright is dedicated to the Public Domain.
http://creativecommons.org/publicdomain/zero/1.0/
If you like this, you should donate to Peter O.
at: http://upokecenter.com/d/
 */

import com.upokecenter.util.*;

  interface ICBORNumber {

    boolean IsPositiveInfinity(Object obj);

    boolean IsInfinity(Object obj);

    boolean IsNegativeInfinity(Object obj);

    boolean IsNaN(Object obj);

    double AsDouble(Object obj);

    Object Negate(Object obj);

    Object Abs(Object obj);

    ExtendedDecimal AsExtendedDecimal(Object obj);

    ExtendedFloat AsExtendedFloat(Object obj);

    ExtendedRational AsExtendedRational(Object obj);

    float AsSingle(Object obj);

    BigInteger AsBigInteger(Object obj);

    long AsInt64(Object obj);

    boolean CanFitInSingle(Object obj);

    boolean CanFitInDouble(Object obj);

    boolean CanFitInInt32(Object obj);

    boolean CanFitInInt64(Object obj);

    boolean CanTruncatedIntFitInInt64(Object obj);

    boolean CanTruncatedIntFitInInt32(Object obj);

    int AsInt32(Object obj, int minValue, int maxValue);

    boolean IsZero(Object obj);

    int Sign(Object obj);

    boolean IsIntegral(Object obj);
  }
